### PR TITLE
[CPP] Compile fixes for ARM cpus

### PR DESCRIPTION
**Pull request by merlimat**
### Motivation

Fixed compilation on ARM CPUs: 
 * Disabled SSE based CRC32C 
 * Fixed some variables initialization

### CODE REVIEW

**Comment by aahmed-se** [comment]
add github action to check compilation on arm

File: pulsar-client-cpp/perf/RateLimiter.h [hunk 1]
```diff
@@ -50,7 +50,7 @@ RateLimiter::RateLimiter(double rate)
         : interval_(std::chrono::microseconds((long)(1e6 / rate))),
           storedPermits_(0.0),
           maxPermits_(rate),
-          nextFree_() {
```
Comment (ivankelly) [inline pulsar-client-cpp/perf/RateLimiter.h]: These last two changes look completely unrelated. Add something to the commit message to at least explain them.

File: pulsar-client-cpp/lib/checksum/crc32c_sse42.cc [hunk 2]
```diff
@@ -228,3 +239,12 @@ uint32_t crc32c(uint32_t init, const void *buf, size_t len, const chunk_config *
     DEBUG_PRINTF1("crc = 0x%08x\n", crc);
     return crc;
 }
+
+#else // ! BOOST_ARCH_X86_64
```
Comment (ivankelly) [inline pulsar-client-cpp/lib/checksum/crc32c_sse42.cc]: Does this mean that arm clients can't use crc32c at all?

Comment (merlimat) [reply pulsar-client-cpp/lib/checksum/crc32c_sse42.cc]: This is only for the SSE 4.2 based implementation, otherwise we fallback to the pure C implementation.

File: pulsar-client-cpp/perf/RateLimiter.h (hunk 1 above)
Comment (merlimat) [reply pulsar-client-cpp/perf/RateLimiter.h]: These are to correct some initialization differences with the compiler/arch combinations. Without these, the simple perf producer was not working on arm with GCC 6.3.

It's mentioned (although briefly) in the description: 

> Fixed some variables initialization

**Comment by merlimat** [comment]
> add github action to check compilation on arm

@aahmed-se Once we get a stable build on github actions for x86_64, then we can add a new build for arm.

**Comment by aahmed-se** [comment]
@merlimat we have a stable build for x86
https://github.com/apache/pulsar/pull/6129/checks?check_run_id=405497005
